color/src/main/java/com/afollestad/materialdialogs/color/ColorGridAdapter.kt

```diff
@@ -107,10 +107,12 @@ internal class ColorGridAdapter(
     if (initialSelection != null) {
       selectedTopIndex = colors.indexOfFirst { it == initialSelection }
       if (selectedTopIndex == -1 && subColors != null) {
-        for (section in subColors) {
-          selectedSubIndex = section.indexOfFirst { it == initialSelection }
+        for (index in 0 until subColors.size - 1) {
```

Comment:
You can avoid `size - 1` by using

```kotlin
for (index in 0..subColors.size) {
    ...
}
```

Reply:
Just realised, that this is wrong, probably you meant `until` ;-) `...` is including the last value